File: Client_Root/Client/Assets/Scripts/Network/Messages/ToServer/ReadyForStartToS.cs
using System.Text;

public class ReadyForStartToS : IMessage
{
    public ulong m_nPlayerNumber;     //  json field name : PlayerNumber

    public ushort GetID()
    {
        return (ushort)Messages.ReadyForStartToS_ID;
    }

    public byte[] Serialize()
    {
        JSONObject jsonObj = new JSONObject(JSONObject.Type.OBJECT);

        JSONHelper.AddField(jsonObj, "PlayerNumber", m_nPlayerNumber);

        return Encoding.Default.GetBytes(jsonObj.Print());
    }

    public bool Deserialize(byte[] bytes)
    {
        JSONObject jsonObj = new JSONObject(Encoding.UTF8.GetString(bytes));

        if(!JSONHelper.GetField(jsonObj, "PlayerNumber", ref m_nPlayerNumber)) return false;

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ReadyForStartToS.cpp
#include "stdafx.h"
#include "ReadyForStartToS.h"
#include "RoomMessageDefines.h"
#include "../../CommonSources/Message/JSONHelper.h"


ReadyForStartToS::ReadyForStartToS()
{
	m_buffer = new GenericStringBuffer<UTF8<>>();
	m_writer = new Writer<StringBuffer, UTF8<>>(*m_buffer);
}

ReadyForStartToS::~ReadyForStartToS()
{
	delete m_buffer;
	delete m_writer;
}

unsigned short ReadyForStartToS::GetID()
{
	return (unsigned short)Messages::ReadyForStartToS_ID;
}

const char* ReadyForStartToS::Serialize()
{
	Document document;
	document.SetObject();

	JSONHelper::AddField(&document, "PlayerIndex", m_nPlayerIndex);

	m_buffer->Clear();
	document.Accept(*m_writer);

	return m_buffer->GetString();
}

bool ReadyForStartToS::Deserialize(const char* pChar)
{
	Document document;
	document.Parse<0>(pChar);
	if (!document.IsObject())
	{
		return false;
	}

	if (!JSONHelper::GetField(&document, "PlayerIndex", &m_nPlayerIndex)) return false;

	return true;
}